spec/controllers/topologies_controller_spec.rb

```diff
@@ -2,275 +2,144 @@
+  #  }
+
+  #  describe "removing a child should spawn an error" do
+  #    let(:graph){
+  #      {'name' => "hv network", 'children' => [
+  #        { 'name' => 'mv network', 'children' => [
+  #          {'name' => "lv1", 'stakeholder' => 'customer'},
+  #          {'name' => "lv2", 'stakeholder' => 'customer'}
+  #        ]}
+  #      ]}
+  #    }
+
+  #    let(:clone_graph){
+  #      {'name' => "hv network", 'children' => [
+  #        { 'name' => 'mv network', 'children' => [
+  #          {'name' => "lv1", 'stakeholder' => 'customer'}
+  #        ]}
+  #      ]}
+  #    }
+
+  #    let!(:another_testing_ground){
+  #      FactoryGirl.create(:testing_ground, user: user, topology: topology)
+  #    }
+
+  #    it "doesn't clone the graph due to removal of a child" do
+  #      patch :clone, format: :js, id: topology.id, topology: {
+  #        graph: JSON.dump(clone_graph)
+  #      }
+
+  #      expect(topology.count).to eq(1)
+  #    end
+  #  end
+
+  #  describe "with other les's" do
+  #    let(:graph){
+  #      {'name' => "hv network", 'children' => [
+  #        { 'name' => 'mv network', 'children' => [
+  #          {'name' => "lv1", 'stakeholder' => 'customer'},
+  #          {'name' => "lv2", 'stakeholder' => 'customer'}
+  #        ]}
+  #      ]}
+  #    }
+
+  #    let!(:another_testing_ground){
+  #      FactoryGirl.create(:testing_ground, user: user, topology: topology)
+  #    }
+
+  #    it 'creates a 2nd topology' do
+  #      perform_clone
+
+  #      expect(Topology.count).to eq(2)
+  #    end
+
+  #    it 'makes a cloned topology private' do
+  #      perform_clone
+
+  #      expect(Topology.last.public).to eq(false)
+  #    end
+
+  #    it "adds the clone number to the topology name" do
+  #      perform_clone
+
+  #      expect(Topology.last.name).to eq("Topology - clone #1")
+  #    end
+  #  end
+
+  #  describe "with no other les's" do
+  #    let(:graph){
+  #      {'name' => "hv network", 'children' => [
+  #        { 'name' => 'mv network', 'children' => [
+  #          {'name' => "lv1", 'stakeholder' => 'customer'},
+  #          {'name' => "lv2", 'stakeholder' => 'customer'}
+  #        ]}
+  #      ]}
+  #    }
+
+  #    it 'does not create a 2nd topology' do
+  #      perform_clone
+
+  #      expect(Topology.count).to eq(1)
+  #    end
+  #  end
+
+  #  describe "with a wrong topology" do
+  #    let(:graph){
+  #      "---\r\nname: hv network\r\nchildren:\r\n- name: mv network\r\n  children:\r\n  - name: \"lv #1\"\r\n  - name: \"lv #2\"\r\n  \r\n  \r\nfailure\r\n  - name: \"lv #3\"\r\n"
+  #    }
+
+  #    it 'does not create a 2nd topology due to errors' do
+  #      perform_clone
+
+  #      expect(Topology.count).to eq(1)
+  #    end
+  #  end
+
+  #  describe "with no topology" do
+  #    let(:graph){ "" }
+
+  #    it 'does not create a 2nd topology due to errors' do
+  #      perform_clone
+
+  #      expect(Topology.count).to eq(1)
+  #    end
+  #  end
+  #end
```

Comment:
Yes, I'm getting rid of the `Cloner` and probably going to write something different for the sake of https://github.com/quintel/etmoses/issues/1499